### PR TITLE
[RNMobile] FlatList inside the bottom-sheet

### PR DESCRIPTION
**Pull request by dratwas**
## Description
In this PR i added a possibility to use FlatList inside Bottom-sheet w/o losing the swipe to close functionality (iOS) and w/o losing the optimization of FLatList.

Because we used the ScrollView to wrap content inside the BottomSheet we closed the optimization of FlatList. In this PR I added a prop called `isChildrenScrollable` and add ScrollView wrapper depends on it. I pass also all props that need to be passed to the nested FlatList by BottomSheetConsumer.

I demonstrated it on the inserter menu since it was a FlatList with `scrollEnabled` set to false.

## How has this been tested?
- Run app in develop mode
- Open inserter
- There should not be the warning `WARN  VirtualizedLists should never be nested inside plain ScrollViews with the same orientation - use another VirtualizedList-backed container instead.
` in the console

## Types of changes
Add possibility to use FlatList inside bottom-sheet

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by mkevins — COMMENTED** [review]
I tested this via the steps described on Android (Pixel 3a) and iOS (via simulator iPhone 11 Pro), and the warning is no longer logged. I tested out the bottomsheet via the inserter, as well as some of the more complex flows, such as the color settings in button, and mostly everything works as expected.

Also, I've tested this via inclusion of the changeset in a draft PR here: https://github.com/WordPress/gutenberg/pull/23922, and it seems to be working there as well.

However, I noticed one odd regression in the scroll behavior of the inserter: when the drag start happens between the buttons, the gesture is lost:

<details>
<summary>Swipe gesture is lost between buttons</summary>


</details>

I also encountered a subtle unrelated bug while testing (also observed on master), noting here since there is related work underway with navigation:

In the colors subsheet, after using the "soft" back button (the one in the subsheet header), two button presses are required to dismiss the main modal via the hardware back button:

<details>
<summary>Back button requires two presses after using "soft" back button in subsheet</summary>


</details>

**Comment by dratwas** [comment]
Hey @mkevins thanks for your review :)
> However, I noticed one odd regression in the scroll behavior of the inserter: when the drag start happens between the buttons, the gesture is lost:

I already fixed it, could you please check it one more time?

>In the colors subsheet, after using the "soft" back button (the one in the subsheet header), two button presses are required to dismiss the main modal via the hardware back button:

Thanks for reporting, i already reimplemented the back button functionality and it seems to work on the navigation PR

**Review by mkevins — APPROVED** [review]
Tested again after the latest fix, and the gesture is no longer lost between buttons :tada: . LGTM!

**Comment by lukewalczak** [comment]
I've noticed one regression in an inserter items layout, which should be left-aligned. To illustrate I attached screenshots:

before | after changes
--- | ---
Screenshot 2020-07-23 at 09 18 19 | Screenshot 2020-07-23 at 09 18 02

**Comment by dratwas** [comment]
Thanks @lukewalczak for your report! I already fixed that issue. I also added the correct padding bottom to the inserter menu. Could you please check it one more time? :)

File: packages/components/src/mobile/bottom-sheet/index.native.js [hunk 1]
```diff
@@ -246,6 +246,15 @@ class BottomSheet extends Component {
 		return onClose();
 	}
 
+	getContentStyle() {
+		const { safeAreaBottomInset } = this.state;
+		return {
+			flexGrow: 1,
+			paddingBottom:
```
Comment (lukewalczak) [inline packages/components/src/mobile/bottom-sheet/index.native.js:253]:
```suggestion
			paddingBottom:
				( safeAreaBottomInset || 0 ) +
				styles.scrollableContent.paddingBottom,
```

**Review by lukewalczak — APPROVED** [review]

**Pull request merged**